### PR TITLE
table service improvements and docs update

### PR DESCRIPTION
**Pull request by mario-subo**
image
image

### CODE REVIEW

File: projects/ng-components/lib/table/table-td/table-td.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <ng-container [ngSwitch]="valueType">
-    <span *ngSwitchCase="'string'">{{ row }}</span>
-    <span *ngSwitchCase="'number'">{{ row | number }}</span>
-    <span *ngSwitchCase="'datetime'">{{ row | date: dateFormat + " HH:mm:ss" }}</span>
-    <span *ngSwitchCase="'date'">{{ row | date: dateFormat }}</span>
-    <span *ngSwitchCase="'custom-html'" [innerHtml]="row | safeHtml"></span>
+    <span *ngSwitchCase="'string'"> {{ row || "–" }}</span>
```
Comment (omarVodiaK) [inline projects/ng-components/lib/table/table-td/table-td.component.html:2]: "–" should be possible to configure, not all the time its a simple string, sometimes it should be empty or in my case it has to be a tooltip, would it be possible to add some extra Input to enable this kind of feature ?

Comment (mario-subo) [reply projects/ng-components/lib/table/table-td/table-td.component.html:2]: Would `custom-html` suffice for your example or do you want to be able to set your own string to display if the row is empty?

File: projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+export const tableServiceSortSnippet: string = `
+@Component({
+    // Bind the sort properties to the template
+    template: \`<sebng-table [sortInfo]="sortInfo$ | async" (sortClicked)="changeSort($event)" ... ></sebng-table>\` 
+})
+export class MyTableComponent {
+    ...
+
+    sortInfo$;
```
Comment (kherP) [inline projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts:49]: i know this is just a snippet, but wondering if we should include the type to promote good programming practices?

File: projects/docs/src/app/docs-wrapper/pages/table-page/table-page.component.ts [hunk 3]
```diff
@@ -1,75 +1,121 @@
...
+            });
+        } else {
+            this.tableService.registerDatasource(this.data, {
+                columns: this.getColumnsFromDropdownList(this.selectedColumnsDropdownList),
+                types: this.types,
+            });
+        }
+    }
+
+    getDropdownItemByColumnName = (col: string) => {
```
Comment (yousifalraheem) [inline projects/docs/src/app/docs-wrapper/pages/table-page/table-page.component.ts:114]: Put the return type of this method

File: projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts (hunk 2 above)
Comment (yousifalraheem) [reply projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts:49]: yep 👍

File: projects/docs/src/app/docs-wrapper/pages/table-page/table-page.component.ts (hunk 3 above)
Comment (mario-subo) [reply projects/docs/src/app/docs-wrapper/pages/table-page/table-page.component.ts:114]: done ✔️

File: projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts (hunk 2 above)
Comment (mario-subo) [reply projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts:49]: done

File: projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import { TableService, TableHeaderListItem, TableServiceDataAndHandlers } from "@sebgroup/ng-components/table";
+
+@Component({
+    selector: "app-my-table",
+    template: \`
+    <sebng-table
+        [rows]="rows$ | async"
+        [headerList]="headerList$ | async"
+    ></sebng-table>
+    \`,
```
Comment (yousifalraheem) [inline projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts:27]: why are you escaping the tick?

File: projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts [hunk 5]
```diff
@@ -54,7 +54,7 @@ export class MyTableComponent {
 
 export const tableServiceSortSnippet: string = `
 ...
-import { TableService, TableConfig, SortInfo, TableHeaderListItem, TableServiceDataAndHandlers } from "@sebgroup/ng-components/table";
+import { ..., TableConfig, SortInfo } from "@sebgroup/ng-components/table";
```
Comment (yousifalraheem) [inline projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts:57]: What's this?? `...`??

Comment (mario-subo) [reply projects/docs/src/app/docs-wrapper/pages/table-page/notes-snippets.ts:57]: removed

**Pull request merged**